src/main/java/jenkins/metrics/api/MetricProvider.java

```diff
@@ -51,6 +51,16 @@ protected static MetricSet metrics(Map.Entry<String, Metric>... metrics) {
         return new FixedMetricSet(result);
     }
 
+    protected static MetricSet metrics(Map<String, Metric> metrics) {
+        final Map<String, Metric> result = new LinkedHashMap<String, Metric>(metrics.size());
```

Comment:
Nope you'd need to do `return metrics(metrics.entrySet().toArray(new Map.Entry<String,Metric>[metrics.size()]));`

Which would be more expensive